core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/client/withClient.ts

```diff
@@ -0,0 +1,81 @@
+import {
+    patchState,
+    signalStoreFeature,
+    type,
+    withComputed,
+    withMethods,
+    withState
+} from '@ngrx/signals';
+
+import { computed } from '@angular/core';
+
+import { DotPageApiParams } from '../../../services/dot-page-api.service';
+import { UVEState } from '../../models';
+
+export interface clientRequestProps {
+    params: Partial<DotPageApiParams>;
+    query: string;
+}
+
+export interface ClientConfigState {
+    isClientReady: boolean;
+    clientRequestProps: clientRequestProps;
+}
+
+const initialState: ClientConfigState = {
+    isClientReady: false,
+    clientRequestProps: {
+        params: {},
+        query: ''
+    }
+};
+
+/**
+ * Add methods to handle client configuration
+ * The Client represents the page host. In the context of self-hosted pages, dotCMS acts as the client.
+ *
+ * @description This feature is used to handle the client configuration
+ * @export
+ * @return {*}
+ */
+export function withClient() {
+    return signalStoreFeature(
+        {
+            state: type<UVEState>()
+        },
+        withState<ClientConfigState>(initialState),
+        withComputed((store) => {
+            return {
+                $clientRequestProps: computed<clientRequestProps>(() => {
+                    const { query, params } = store.clientRequestProps();
+
+                    return {
+                        query,
+                        params: {
+                            ...store.params(),
+                            ...params
+                        }
+                    };
+                })
+            };
+        }),
```

Comment:
Computed props should be a reflection of the UI and here you are just accessing the state. Do we need this computed? I think we can do the same by just accessing the store directly.

In the place we call the PageAPI we already have the params, we can do the merge there. I think we should separate the params of the editor from the params the user sends from the client.

See comments below.

Reply:
That makes sense, that was my first approach but we could use the computed for methods unrelated to the UI